R/dependency.R

```diff
@@ -0,0 +1,104 @@
+#' Add an external JavaScript asset as a dygraph dependency
+#'
+#' @param dygraph Dygraph to add dependency to
+#' @param type Type of dependency (valid options are: 'plugins', 'plotter' or 'dataHandler')
+#' @param name Name of dependency
+#' @param path Path to dependency JavaScript file
+#' @param version Dependency version (e.g. version of package which provides the dependency)
+#' @param all.files Whether all files under the path directory are dependency files
+#'
+#' @return A dygraph with the specified dependency added.
+#'
+#' @importFrom htmltools htmlDependency
+#'
+#' @export
+dyDependency <- function(dygraph, type, name, path, version = "1.0", all.files = FALSE) {
```

Comment:
I think the most flexible thing to do here is actually going to be taking just two arguments:

```r
dyDependency <- function(dygraph, dependency)
```

Then allow the user to use `htmltools::htmlDependency` to construct any dependency they wish (as things currently stand we intermediate that interface so e.g. you can't add CSS or meta tags).